google/earthenterprise
# 566 postgres install rpm data upgrade

## Verification Steps

1. Create a VM (RHEL or CentOS) with GEE Server 5.2.0 installed using install_server.sh script
2. Publish one or more globes with search tabs
3. Verify the globe(s) load using Earth EC
4. Do a search or two and take note of the results
5. Take a snapshot for ease of going back to this state
6. Remove any old /tmp/fusion_os_install directories
7. Build, create RPM files, and install the feature branch
8. The geserver service should start without issues
9. Log into the admin console and verify that the database(s) are still present
10. Verify the globe(s) load using Earth EC with "disable cache" set
11. Do the same searches as before and make sure the results are the same
12. Go back to step 7. You can skip build, and just run stage install, and create RPM files again. There should be additional RPM files with a new version.
13. Continue with the same verification once more to verify that the install upgrade also works with PostgreSQL 9.6 installed

Optional (because there are known issues with updating RPM files from master)

1. Install RPM files built from master
1. Run steps 2-11 from above

Issues you may encounter:
1. When the RPMs are upgraded they will remove the GEE users and the service configuration
    * Solution: reinstall the RPM files `sudo yum reinstall opengee*<version>*.rpm"
1. The opengee-common RPM may fail to be removed with an error such as: 
    `opengee-common-5.2.0-1.el6.x86_64 was supposed to be removed but is not!`
    
    * Solution: remove the package manually and disable running the script: `sudo yum remove --setopt=tsflags=noscripts "opengee-common-5.2.0-1.el6.x86_64" ` (The exact package name should match the one in the error message)

I merged the latest master into this branch and postgres_upgrade for testing. Still targeting postgres_upgrade for now so these changes can be reviewed independently.
